# Cleansing Data Youtube Rewind 2021

#### Import Library

In [1]:
from dotenv import load_dotenv
from pymongo import MongoClient
import os
import pandas as pd

#### Import url dari .env

In [2]:
# Memuat value dari file .env
load_dotenv()

mongodb_url = os.getenv('URL_SANDY')
local_url = os.getenv('URL_LOCAL')

In [3]:
database_name = "youtube_rewind_indonesia"
collection_from = "collect_2021"
collection_name = "cleansing_2021"

#### Import dataset dari MongoDB

In [4]:
# Koneksi ke MongoDB
client = MongoClient(local_url)

# Pilih database dan koleksi
db = client[database_name]
collection = db[collection_from]

# Query data dari koleksi
cursor = collection.find()

# Konversi cursor ke list dan kemudian ke DataFrame
data = list(cursor)
df = pd.DataFrame(data)

# Menutup koneksi
client.close()

In [5]:
df

_id                 channelId      videoId  \
0      6651666d9da7c08a9d7f3bbe  UC-z1ubmS-iHMVr83EEwRS2g  IPZO85HFM4w   
1      6651666d9da7c08a9d7f3bbf  UC-z1ubmS-iHMVr83EEwRS2g  IPZO85HFM4w   
2      6651666d9da7c08a9d7f3bc0  UC-z1ubmS-iHMVr83EEwRS2g  IPZO85HFM4w   
3      6651666d9da7c08a9d7f3bc1  UC-z1ubmS-iHMVr83EEwRS2g  IPZO85HFM4w   
4      6651666d9da7c08a9d7f3bc2  UC-z1ubmS-iHMVr83EEwRS2g  IPZO85HFM4w   
...                         ...                       ...          ...   
69295  6651673e9da7c08a9d804a6d  UC-z1ubmS-iHMVr83EEwRS2g  IPZO85HFM4w   
69296  6651673e9da7c08a9d804a6e  UC-z1ubmS-iHMVr83EEwRS2g  IPZO85HFM4w   
69297  6651673e9da7c08a9d804a6f  UC-z1ubmS-iHMVr83EEwRS2g  IPZO85HFM4w   
69298  6651673e9da7c08a9d804a70  UC-z1ubmS-iHMVr83EEwRS2g  IPZO85HFM4w   
69299  6651673e9da7c08a9d804a71  UC-z1ubmS-iHMVr83EEwRS2g  IPZO85HFM4w   

                                             textDisplay  \
0      REWIND INDONESIA 2021.<a href="http://www.yout...   
1                                                      .   
2                                           ❤❤❤❤❤😊😊😊😢😢😢😢   
3                                                ❤❤❤❤❤❤❤   
4                  easter egg tapi rewind indonesia 2021   
...                                                  ...   
69295                                         MERINDING👏   
69296                                            KERENN🫂   
69297                                         Keyennnnnn   
69298                                         MERINDING!   
69299  Hallo?? Kok keren banget yah??? Indonesia rewi...   

                                            textOriginal  \
0      REWIND INDONESIA 2021.#RI2021\nYuk bikin video...   
1                                                      .   
2                                           ❤❤❤❤❤😊😊😊😢😢😢😢   
3                                                ❤❤❤❤❤❤❤   
4                  easter egg tapi rewind indonesia 2021   
...                                                  ...   
69295                                         MERINDING👏   
69296                                            KERENN🫂   
69297                                         Keyennnnnn   
69298                                         MERINDING!   
69299  Hallo?? Kok keren banget yah??? Indonesia rewi...   

            authorDisplayName  \
0          @IndonesianCreator   
1             @remajagpdi5046   
2      @nazwaauliarahma-vf9kb   
3      @nazwaauliarahma-vf9kb   
4      @irfanal-mushoddiq9832   
...                       ...   
69295              @diwoniest   
69296         @tiaradwisftri0   
69297          @4maskentir435   
69298          @asepriyan4031   
69299         @yanayulinda865   

                                   authorProfileImageUrl  \
0      https://yt3.ggpht.com/fO77gm9qrRYfPRp2NDPodQ9i...   
1      https://yt3.ggpht.com/EnI2iOSzqpzxeq1ahuXlLdaJ...   
2      https://yt3.ggpht.com/oCmwqF-kksOmEGAM9DJFIPM-...   
3      https://yt3.ggpht.com/oCmwqF-kksOmEGAM9DJFIPM-...   
4      https://yt3.ggpht.com/ytc/AIdro_k1jPTiuIIJX6oX...   
...                                                  ...   
69295  https://yt3.ggpht.com/-SSCQ7fviTkaF9bzQdFL4okj...   
69296  https://yt3.ggpht.com/aYTn6ZaxW07D_ba9BW33MpKB...   
69297  https://yt3.ggpht.com/ytc/AIdro_kruly7M0IjKnbk...   
69298  https://yt3.ggpht.com/ytc/AIdro_mQWgZog6XaFhyP...   
69299  https://yt3.ggpht.com/ytc/AIdro_llddFss8F93p4d...   

                                    authorChannelUrl  \
0          http://www.youtube.com/@IndonesianCreator   
1             http://www.youtube.com/@remajagpdi5046   
2      http://www.youtube.com/@nazwaauliarahma-vf9kb   
3      http://www.youtube.com/@nazwaauliarahma-vf9kb   
4      http://www.youtube.com/@irfanal-mushoddiq9832   
...                                              ...   
69295              http://www.youtube.com/@diwoniest   
69296         http://www.youtube.com/@tiaradwisftri0   
69297          http://www.youtube.com/@4maskentir435   
69298          http

#### Menghapus data text yang duplikat

In [6]:
df = df.drop_duplicates(subset=['textOriginal'])

df.shape

(62441, 14)

#### Menghapus data author yang duplikat

In [7]:
df = df.drop_duplicates(subset=['authorChannelUrl'])

df.shape

(59541, 14)

#### Menghilangkan atribut yang tidak diperlukan

Karena hanya menganalisis komentar, maka atribut yang disisakan hanya _id dan textOriginal

In [8]:
df = df[['_id', 'textOriginal']]
df

_id  \
0      6651666d9da7c08a9d7f3bbe   
1      6651666d9da7c08a9d7f3bbf   
2      6651666d9da7c08a9d7f3bc0   
4      6651666d9da7c08a9d7f3bc2   
5      6651666d9da7c08a9d7f3bc3   
...                         ...   
69294  6651673e9da7c08a9d804a6c   
69295  6651673e9da7c08a9d804a6d   
69296  6651673e9da7c08a9d804a6e   
69297  6651673e9da7c08a9d804a6f   
69299  6651673e9da7c08a9d804a71   

                                            textOriginal  
0      REWIND INDONESIA 2021.#RI2021\nYuk bikin video...  
1                                                      .  
2                                           ❤❤❤❤❤😊😊😊😢😢😢😢  
4                  easter egg tapi rewind indonesia 2021  
5      Anjirrrr , Nonton lagi dan ternyata ada Lord U...  
...                                                  ...  
69294      MasyaAllah keren sekaliiiii i love u🥺🤍🤍🤍🤍!!!!  
69295                                         MERINDING👏  
69296                                            KERENN🫂  
69297                                         Keyennnnnn  
69299  Hallo?? Kok keren banget yah??? Indonesia rewi...  

[59541 rows x 2 columns]

#### Menghapus nilai yang kosong

In [9]:
df = df.dropna()

# Melihat data yang kosong
df.isnull().sum()

_id             0
textOriginal    0
dtype: int64

Dari yang awalnya berjumlah 69300 row berkurang menjadi 59541 row

#### Mengimpan Clean Comments ke MongoDB Local

In [10]:
# Koneksi ke MongoDB
client = MongoClient(local_url)
db = client[database_name]
collection = db[collection_name]

# Konversi DataFrame ke bentuk dictionary
records = df.to_dict(orient='records')

# Menyisipkan data ke MongoDB
collection.insert_many(records)

print("Data berhasil diimpor ke MongoDB Local.")

Data berhasil diimpor ke MongoDB Local.


#### Migrasi dari MongoDB Local ke MongoDB Atlas

In [11]:
# Koneksi ke MongoDB lokal
local_client = MongoClient(local_url)
local_db = local_client[database_name]
local_collection = local_db[collection_name]

# Koneksi ke MongoDB Atlas
atlas_client = MongoClient(mongodb_url)
atlas_db = atlas_client[database_name]
atlas_collection = atlas_db[collection_name]

# Ambil semua dokumen dari koleksi MongoDB lokal
documents = list(local_collection.find())

# Masukkan dokumen ke MongoDB Atlas
if documents:
    atlas_collection.insert_many(documents)
    print(f'{len(documents)} dokumen berhasil dipindahkan dari MongoDB lokal ke MongoDB Atlas.')
else:
    print('Tidak ada dokumen yang ditemukan di MongoDB lokal.')

# Tutup koneksi
local_client.close()
atlas_client.close()

KeyboardInterrupt: 